src/Service/Geocoder.php

```diff
@@ -0,0 +1,97 @@
+<?php
+
+namespace App\Service;
+
+use App\Entity\Address;
+use App\Exception\GeocoderException;
+use Symfony\Contracts\HttpClient\Exception\ClientExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\DecodingExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\RedirectionExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\ServerExceptionInterface;
+use Symfony\Contracts\HttpClient\Exception\TransportExceptionInterface;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+final class Geocoder implements GeocoderInterface
+{
+    public function __construct(
+        private readonly HttpClientInterface $client,
+    ) {
+    }
+
+    public function encode(Address $address): array
```

Comment:
We should use some cache around this function since it will likely be called multiple times with the same address data.

Reply:
Cache added